### PR TITLE
Removing GIDs from the spec

### PR DESCRIPTION
**Pull request by hernando**
This change mainly affects the spec for reports and node sets.
Some datasets have been renamed to use plural for consistency.

### CODE REVIEW

**Comment by hernando** [comment]
Addresses #46

File: docs/SONATA_DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -787,7 +787,7 @@ Where to find the hoc templates for the edges:
            "templates": "$COMPONENT_DIR/hoc_templates",
         },
 
-The network is defined by nodes and edges. In the example below, a V1 model is being simulated (with recurrent connections) that receives input from virtual LGN source nodes. Each population of nodes should contain "nodes" and “node_types” while each population of edges should “edges”, “edge_types”.  Gids are assigned to nodes in advance (using another tool) or during the simulation, depending on implementation, and are global across all  populations in the network (see the (gid mapping)[#gid_mapping] section for details)
```
Comment (arsenius7) [inline docs/SONATA_DEVELOPER_GUIDE.md]: This paragraph is a good example where new line per each sentence would simplify reading diff; we can go through formatting issues in a separate change.

File: docs/SONATA_DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -1251,55 +1244,65 @@ provides a different file name).
 
 #### <a name="spike_file"></a>Spike file
 
-Spikes from all cells will be stored in a single HDF5 file that contains (gid, spike time) pairs in separate datasets. These datasets may be unsorted, sorted by gid or sorted by spike time. The gids are not to be confused with node_ids from populations, see below for details about gid to node_id and population mapping.
+Spikes from all cells will be stored in a single HDF5 file that contains
+(population, node id, spike time) pairs in separate datasets. These datasets
```
Comment (arsenius7) [inline docs/SONATA_DEVELOPER_GUIDE.md]: `(node_id, spike time)` pairs in separate datasets?

File: docs/SONATA_DEVELOPER_GUIDE.md [hunk 3]
```diff
@@ -1251,55 +1244,65 @@ provides a different file name).
...
+* **/spikes** (group): contains one or more groups of per population spikes.
+* **/spikes/<population_name>** (group), attributes:
+    - **sorting** (dtype: enum, optional): It can take one of these
+    values: `none`, `by_id`, `by_time`. All the datasets below are sorted using
+    the datasets specified by the attribute. When sorting `by_id`, spikes are
+    sorted by `node_id`; spikes from the same node are expected to be also
+    sorted by timestamp as secondary key. When sorting `by_time`, spikes with
+    the same timestamp may be in any order. If missing, no sorting can be
+    assumed.
+* **/spikes/<poulation_name>/timestamps** (dytpe: double, shape: N spikes),
```
Comment (arsenius7) [inline docs/SONATA_DEVELOPER_GUIDE.md]: a typo, po*p*ulation

File: docs/SONATA_DEVELOPER_GUIDE.md (hunk 1 above)
Comment (hernando) [reply docs/SONATA_DEVELOPER_GUIDE.md]: Yes. In general I do line wrapping, which is already better, but I didn't touch this line. In any case, you should try using --color-words in the command line, it does a much better job than Github no matter how paragraphs are split.

File: docs/SONATA_DEVELOPER_GUIDE.md [hunk 4]
```diff
@@ -1238,9 +1233,7 @@ node_id mapping is specified according to [description below](#gid_mapping).
             "model_type": "point",
             "node_id": [1, 2, 3, 5, 7, 9, ...]
         }
-        "layer4": {
-            "gids": [1, 2, 3, 4, 5, ...]
-        },
+        "combined": ["bio_layer45", "V1_point_prime"]
```
Comment (kaeldai) [inline docs/SONATA_DEVELOPER_GUIDE.md]: Is "combined" a new node-set?

File: docs/SONATA_DEVELOPER_GUIDE.md (hunk 1 above)
Comment (kaeldai) [reply docs/SONATA_DEVELOPER_GUIDE.md]: There's a button at the top of the diff for "display the rich diff" which I find much easier for looking at the diff.

File: docs/SONATA_DEVELOPER_GUIDE.md (hunk 4 above)
Comment (hernando) [reply docs/SONATA_DEVELOPER_GUIDE.md]: It's just a name for an example of a compound node set, as there was example of this before. I couldn't think of a better name for it.

File: docs/SONATA_DEVELOPER_GUIDE.md (hunk 1 above)
Comment (hernando) [reply docs/SONATA_DEVELOPER_GUIDE.md]: I didn't know that one. For Markdown it really makes it a lot easier to read the differences. However, it doesn't work very well for HTML elements within the Markdown rendering (e.g. the table in which I modified one word in just one cell appears as if it was entirely replaced by a new one).

**Pull request merged**